projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php

```diff
@@ -392,6 +413,79 @@ public function get_locale() {
 		return $wpcom_locale;
 	}
 
+	/**
+	 * Get Jetpack locale name.
+	 *
+	 * @param  string $slug Locale slug.
+	 * @return string Jetpack locale.
+	 */
+	public function get_jetpack_locale( $slug = '' ) {
+		if ( ! class_exists( 'GP_Locales' ) ) {
+			if ( defined( 'JETPACK__GLOTPRESS_LOCALES_PATH' ) && file_exists( JETPACK__GLOTPRESS_LOCALES_PATH ) ) {
+				require JETPACK__GLOTPRESS_LOCALES_PATH;
+			}
+		}
+
+		if ( class_exists( 'GP_Locales' ) ) {
+			$jetpack_locale_object = GP_Locales::by_field( 'slug', $slug );
+			if ( $jetpack_locale_object instanceof GP_Locale ) {
+				$jetpack_locale = $jetpack_locale_object->wp_locale;
```

Comment:
Seems not all `GP_Locales` have a `wp_locale` property defined. Should we include a safeguard check?

Reply:
> Default the locale to English

Maybe this? That's aligned with the discussion on [this related issue](https://github.com/Automattic/wp-calypso/issues/35026). Seems that all the languages not supported in WP Admin shouldn't appear in Calypso.

Reply:
Thanks, that makes sense. I made the change here: https://github.com/Automattic/jetpack/pull/19525/commits/fa5ad064d9c16350968abbb3b6d99d71ca816e7d